File: exercise/kotlin/day24/src/test/kotlin/ToyBuilder.kt
import arrow.core.getOrElse
import com.github.javafaker.Faker
import domain.Toy

class ToyBuilder private constructor(private val stock: Int) {

    companion object {
        fun toysInStock(stock: Int) = ToyBuilder(stock)
    }

    fun build(): Toy =
        Toy.create(Time.TimeProvider, Faker().lorem().word(), stock)
            .getOrElse { throw RuntimeException("Toy creation failed") }
}

File: exercise/scala/day24/src/test/scala/usecases/ToyBuilder.scala
package usecases

import com.github.javafaker.Faker
import domain.Toy

class ToyBuilder private(private val stock: Int) {
  def build(): Toy =
    Toy.create(Time.timeProvider, Faker().lorem().word(), stock)
      .getOrElse(throw new RuntimeException("Toy creation failed"))
}

object ToyBuilder {
  def toysInStock(stock: Int): ToyBuilder = new ToyBuilder(stock)
}